credsweeper/app.py

```diff
@@ -277,31 +278,35 @@ def file_scan(self, content_provider: ContentProvider) -> List[Candidate]:
                         data = diff
                 else:
                     logger.warning(f"Content provider {type(content_provider)} does not support deep scan")
+
                 if data:
+                    new_size_limit = self.config.size_limit if self.config.size_limit else RECURSIVE_SCAN_LIMITATION
+                    new_size_limit -= len(data)
                     data_provider = DataContentProvider(data=data,
```

Comment:
What about make a new class structure regarding scan?
I think create scan directory for those scan methods and put all of scanning methods not only extract methods like below. (ex. `__zip_scan()`, `__tar_scan()`, and so on..)

Reply:
Well, with the new structure i added additionaly optional pathlib.Path and ioBytesIO for our providers to use them in memory only.